### PR TITLE
Fix NgDoCheck doc mistake(#48140)

### LINKED ISSUE #48140: ngDoCheck hook has a bit misleading documentation
### Describe the problem that you experienced

Hello,

in my opinion the [documentation regarding the `DoCheck` hook](https://angular.io/api/core/DoCheck) is a bit misleading: 
*A lifecycle hook that invokes a custom change-detection function for a directive, **in addition to the check performed by the default change-detector.***

Text in bold implies that the default check was performed (for the component from which the hook is called), when in fact it doesn't need to be the case. If we have a component with the OnPush change detection strategy, its `ngDoCheck` hook is called from its parent, despite the fact that the default check was not performed on the component.

The following article sums it up (it's from 2017, I think that many people fell in the trap of believing that ngDoCheck called = change detection performed):
https://medium.com/angular-in-depth/if-you-think-ngdocheck-means-your-component-is-being-checked-read-this-article-36ce63a3f3e5

### Enter the URL of the topic with the problem

_No response_

### Describe what you were looking for in the documentation

_No response_

### Describe the actions that led you to experience the problem

_No response_

### Describe what you want to experience that would fix the problem

_No response_

### Add a screenshot if that helps illustrate the problem

_No response_

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

_No response_

### Provide any additional information here in as much as detail as you can

_No response_

### PR DESCRIPTION
**Pull request by sr5434**
Fixes #48140

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- NA Tests for the changes have been added (for bug fixes / features)
- NA Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #48140

## What is the new behavior?
NA

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/interface/lifecycle_hooks.ts [hunk 1]
```diff
@@ -65,8 +65,7 @@ export interface OnInit {
 }
 
 /**
- * A lifecycle hook that invokes a custom change-detection function for a directive,
- * in addition to the check performed by the default change-detector.
+ * A lifecycle hook that invokes a custom change-detection function for a directive.
```
Comment (pkozlowski-opensource) [inline packages/core/src/interface/lifecycle_hooks.ts:68]: I'm not sure it 100% clarifies the situation. What is going on here is that the `ngDoCheck` is invoked whenever component _usage_ (in a parent template) is checked. Or to phrase it differently - every time component inputs _might_ have changed.

Alternative phrasing could be:

"A lifecycle hook that invokes a custom change-detection function for a directive,
in addition to the check performed by the default change-detector on the input bindings for this directive usage. Developers might use this hook to implement a custom change detection strategy for some of the inputs."

How does it sound?

**Comment by pkozlowski-opensource** [comment]
@sr5434 thnx for the PR - I've suggested [alternative wording](https://github.com/angular/angular/pull/48209#discussion_r1042299939) - WDYT? 

On top of this there some CI check failures (it is mostly about following commit message conventions) that would have to be fixed before we can merge this one.

File: packages/core/src/interface/lifecycle_hooks.ts (hunk 1 above)
Comment (mlc-mlapis) [reply packages/core/src/interface/lifecycle_hooks.ts:68]: @pkozlowski-opensource I would recommend explicitly mentioning two things, the first is that the hook is always run, and the second is that it is possible to overwrite even the implicit decision of the OnPush strategy by the ngDoCheck turn.

Comment (lukasmatta) [reply packages/core/src/interface/lifecycle_hooks.ts:68]: @pkozlowski-opensource I agree with the change you suggested, but please note that the misleading information occurs on two more places:

https://github.com/angular/angular/blob/a6849f27af129588091f635c6ae7a326241344fc/packages/core/src/interface/lifecycle_hooks.ts#L75

https://github.com/angular/angular/blob/a6849f27af129588091f635c6ae7a326241344fc/packages/core/src/interface/lifecycle_hooks.ts#L97

I would also consider mentioning the fact that `ngDoCheck` hook is invoked from the parent component (when the change detection runs on the parent component) more explicitly. Although, I am not sure about the precise wording, as I'm not too familiar with the change detection internals (there are maybe some exceptions to this statement).

Comment (mlc-mlapis) [reply packages/core/src/interface/lifecycle_hooks.ts:68]: > change detection runs on the parent component) more explicitly

It's also related to the execution order of the hooks as described in https://angular.io/guide/lifecycle-hooks#lifecycle-event-sequence.

Comment (sr5434) [reply packages/core/src/interface/lifecycle_hooks.ts:68]: > A lifecycle hook that invokes a custom change-detection function for a directive,
> in addition to the check performed by the default change-detector on the input bindings for this directive usage. Developers might use this hook to implement a custom change detection strategy for some of the inputs.

Sounds good!

**Comment by sr5434** [comment]
@pkozlowski-opensource Do my new changes look good?

**Comment by sr5434** [comment]
Thank you for the suggestions. Are my changes ok?

**Comment by sr5434** [comment]
@lukasmatta @mlc-mlapis Is my PR ready to merge?

**Comment by JeanMeche** [comment]
You first need to update your fork and then rebase on main. (per the ci/circleci: setup failure).

**Comment by pkozlowski-opensource** [comment]
This PR is not in the mergable state and there was no movement in it for a while so I'm going to close it for now. Feel free to open a new, fresh PR if you are still interested in doing the changes.

**Pull request closed without merging**